subsys/bootloader/Kconfig

```diff
@@ -17,6 +17,7 @@ config BUILD_S1_VARIANT
 config S1_VARIANT_IMAGE_NAME
 	string
 	default "mcuboot" if MCUBOOT || BOOTLOADER_MCUBOOT
+	default "spm" if !BOOTLOADER_MCUBOOT && IS_SPM
```

Comment:
Nothing wrong, and no changes requested, i'm just being curios.

This is a prompt less string, hence it cannot be configured in conf files.

Why not use a config and select, and then have something like:
```
config S1_VARIANT_SPM
	bool

config S1_VARIANT_MCUBOOT
	bool

config S1_VARIANT_APP
	bool

config S1_VARIANT_IMAGE
	select S1_VARIANT_MCUBOOT if MCUBOOT || BOOTLOADER_MCUBOOT
	select S1_VARIANT_SPM if !BOOTLOADER_MCUBOOT && IS_SPM
	select S1_VARIANT_APP if !BOOTLOADER_MCUBOOT
...
```

I think that would make some testing in CMake easier, if you can simply test for `if(S1_VARIANT_SPM)`.

(Even `S1_VARIANT_IMAGE` could probably be reduced)

Reply:
This looks much better, I will update to follow this convention.